animation_nodes/algorithms/mesh/triangulate_mesh.pyx

```diff
@@ -0,0 +1,353 @@
+
+    cdef unsigned int *oldIndices = polygons.indices.data
+    cdef unsigned int *oldPolyStarts = polygons.polyStarts.data
+
+    cdef PolygonIndicesList newPolygons = PolygonIndicesList(
+                                          indicesAmount = 3 * triAmount,
+                                          polygonAmount = triAmount)
+    cdef unsigned int *newIndices = newPolygons.indices.data
+    cdef unsigned int *newPolyStarts = newPolygons.polyStarts.data
+    cdef unsigned int *newPolyLengths = newPolygons.polyLengths.data
+
+    cdef Vertex* vertexData
+    cdef Vertices verticesData
+    cdef Py_ssize_t polyStart, triangleIndex, polyIndex
+    cdef Py_ssize_t j, index, earIndex, headIndex, currentIndex
+
+    polyIndex = 0
+    triangleIndex = 0
+    for i in range(polygonAmount):
+        polyStart = oldPolyStarts[i]
+        polyLength = oldPolyLengths[i]
+
+        # Initialization of polygon's vertices.
+        verticesData.head = 0
+        verticesData.data = <Vertex*>PyMem_Malloc(polyLength * sizeof(Vertex))
+
+        vertexData = verticesData.data
+        for j in range(1, polyLength - 1):
+            vertexData[j].index = oldIndices[polyStart + j]
+            vertexData[j].location = vertices.data[oldIndices[polyStart + j]]
+            vertexData[j].previous = j - 1
+            vertexData[j].next = j + 1
+
+        index = 0
+        vertexData[index].index = oldIndices[polyStart + index]
+        vertexData[index].location = vertices.data[oldIndices[polyStart + index]]
+        vertexData[index].previous = polyLength - 1
+        vertexData[index].next = 1
+
+        index = polyLength - 1
+        vertexData[index].index = oldIndices[polyStart + index]
+        vertexData[index].location = vertices.data[oldIndices[polyStart + index]]
+        vertexData[index].previous = polyLength - 2
+        vertexData[index].next = 0
+
+        # Project vertices in xy-plane.
+        projectPolygonVertices(verticesData)
+
+        # Check polygon polarity.
+        makeSurePolygonIsCounterClockwise(verticesData)
+
+        # Calculate angle and status of vertices.
+        for j in range(polyLength):
+            vertexData[j].angle = calculateAngleWithSign(vertexData, j)
+            setVertexEarStatus(verticesData, j)
+
+        # Calculate triangle polygon indices.
+        for j in range(polyLength - 3):
+            # Removing an ear which has smallest inner-angle.
+            earIndex = findEarHasMinAngle(verticesData)
+
+            newPolyStarts[triangleIndex] = polyIndex
+            newPolyLengths[triangleIndex] = 3
+            triangleIndex += 1
+
+            newIndices[polyIndex] = vertexData[vertexData[earIndex].previous].index
+            newIndices[polyIndex + 1] = vertexData[earIndex].index
+            newIndices[polyIndex + 2] = vertexData[vertexData[earIndex].next].index
+            polyIndex += 3
+
+            if earIndex == verticesData.head: verticesData.head = vertexData[earIndex].next
```

Comment:
Put this line inside the `removeEarVertex` function.

Reply:
I have tried, it does not work inside the `removeEarVertex` function.

Reply:
Can you show me what you tried?

Reply:
> Put this line inside the `removeEarVertex` function.

Done.